finesse/gui/led_icons.py

```diff
@@ -0,0 +1,90 @@
+"""Class for LED Icons."""
+from importlib import resources
+
+from PySide6.QtCore import QTimer
+from PySide6.QtGui import QImage, QPixmap
+from PySide6.QtWidgets import QLabel
+
+_img_files = resources.files("finesse.gui.images")
+_poll_on_img_data = _img_files.joinpath("poll_on.png").read_bytes()
+_poll_off_img_data = _img_files.joinpath("poll_off.png").read_bytes()
+_alarm_on_img_data = _img_files.joinpath("alarm_on.png").read_bytes()
+_alarm_off_img_data = _img_files.joinpath("alarm_off.png").read_bytes()
+
+_poll_on_img = QImage.fromData(_poll_on_img_data)
+_poll_off_img = QImage.fromData(_poll_off_img_data)
+_alarm_on_img = QImage.fromData(_alarm_on_img_data)
+_alarm_off_img = QImage.fromData(_alarm_off_img_data)
+
+
+class LEDIcon(QLabel):
+    """QLabel object to represent an LED with on/off status."""
+
+    def __init__(self, on_img: QImage, off_img: QImage, is_on: bool = False) -> None:
+        """Creates the LED icon, sets its status and stores corresponding image data.
+
+        Args:
+            on_img (QImage): QImage for LED on state.
+            off_img (QImage): QImage for LED off state.
+            is_on (bool): On/off status of LED.
+        """
+        super().__init__()
+        self._on_img = QImage(on_img)
+        self._off_img = QImage(off_img)
+        if is_on:
+            self._turn_on()
+        else:
+            self._turn_off()
+        self._timer = QTimer()
+        self._timer.timeout.connect(self._turn_off)  # type: ignore
+
+    def _turn_on(self):
+        """Turns the LED on."""
+        self._is_on = True
+        self.setPixmap(QPixmap(self._on_img))
+
+    def _turn_off(self):
+        """Turns the LED off."""
+        self._is_on = False
+        self.setPixmap(QPixmap(self._off_img))
+
+    def _flash(self, duration: int = 250):
+        """Turns the LED on for a specified duration.
+
+        Args:
+            duration (int): Number of milliseconds to keep LED lit for
+        """
+        self._turn_on()
+        self._timer.singleShot(duration, self._turn_off)
+
+
+class PollIcon(LEDIcon):
```

Comment:
I think this and the alarm version should not be subclasses of LEDIcon, but factory methods of LEDicon that create the object with slightly different inputs.

So these two init methods should be converted into two class methods of LEDicon, called constructors. Then, in the tenperature controler, yo use those as:

```python
normalLED = LEDIcon.create_poll_icon()
alarmLED = LEDIcon.create_alarm_icon()
```

In summary, just one class- you just create it in a slightly different way.

Reply:
Thanks! I knew there must be a neater way haha. I think what I've now done achieves this.